mlflow/tracking/_workspace/fluent.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+
+import threading
+from typing import Callable, TypeVar
+
+from mlflow.entities.workspace import Workspace
+from mlflow.exceptions import MlflowException, RestException
+from mlflow.protos import databricks_pb2
+from mlflow.protos.databricks_pb2 import FEATURE_DISABLED
+from mlflow.store.workspace.abstract_store import WorkspaceNameValidator
+from mlflow.tracking.client import MlflowClient
+from mlflow.utils.annotations import experimental
+from mlflow.utils.workspace_context import set_workspace as set_context_workspace
+from mlflow.utils.workspace_utils import DEFAULT_WORKSPACE_NAME
+
+T = TypeVar("T")
+
+_workspace_lock = threading.Lock()
+
+
+def _workspace_client_call(func: Callable[[MlflowClient], T]) -> T:
+    client = MlflowClient()
+    try:
+        return func(client)
+    except RestException as exc:
+        if exc.error_code == databricks_pb2.ErrorCode.Name(databricks_pb2.ENDPOINT_NOT_FOUND):
+            raise MlflowException(
+                "The configured tracking server does not expose workspace APIs. "
+                "Ensure multi-tenancy is enabled.",
```

Comment:
```suggestion
                "Ensure workspace is enabled.",
```

Let's avoid the term 'multi-tenancy' as discussed offline.